pkg/upgradeservice/bootstrap_test.go

```diff
@@ -0,0 +1,135 @@
+package upgradeservice
+
+import (
+	"testing"
+
+	"github.com/replicatedhq/kots/pkg/upgradeservice/types"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestUpdateWithinKubeRange(t *testing.T) {
+	tests := []struct {
+		name           string
+		params         types.UpgradeServiceParams
+		expectError    bool
+		expectedErrMsg string
+	}{
```

Comment:
I don't think we have a good way to test the `bootstrap` and `Serve` methods right now, particularly because of the large volume of side effects the methods seem to have. I've opted for some unit tests here but curious to hear your thoughts:
- https://github.com/replicatedhq/kots/blob/1153828cf4b252ccdc0d05da60c22e810e4c8a99/pkg/upgradeservice/server.go#L23-L29

Reply:
I think a unit test where you have one is the best option at this point.